packages/tool-cache/src/manifest.ts

```diff
@@ -0,0 +1,158 @@
+import * as semver from 'semver'
+import {debug} from '@actions/core'
+
+// needs to be require for core node modules to be mocked
+/* eslint @typescript-eslint/no-require-imports: 0 */
+
+import os = require('os')
+import cp = require('child_process')
+import fs = require('fs')
+
+/*
+NOTE: versions must be sorted descending by version in the manifest
+      this library short circuits on first semver spec match
+
+      platform_version is an optional filter and can be a semver spec or range
+[
+  {
+    "version": "1.2.3",
+    "stable": true,
+    "release_url": "https://github.com/actions/sometool/releases/tag/1.2.3-20200402.6",
+    "files": [
+      {
+        "filename": "sometool-1.2.3-linux-x64.zip",
+        "arch": "x64",
+        "platform": "linux",
+        "platform_version": "18.04"
+        "download_url": "https://github.com/actions/sometool/releases/tag/1.2.3-20200402.6/sometool-1.2.3-linux-x64.zip"
+      },
+    ...
+    ]
+  },
+  ...
+]
+*/
+
+export interface IToolReleaseFile {
+  filename: string
+  // 'aix', 'darwin', 'freebsd', 'linux', 'openbsd',
+  // 'sunos', and 'win32'
+  // platform_version is an optional semver filter
+  // TODO: do we need distribution (e.g. ubuntu).
+  //       not adding yet but might need someday.
+  //       right now, 16.04 and 18.04 work
+  platform: string
+  platform_version?: string
+
+  // 'arm', 'arm64', 'ia32', 'mips', 'mipsel',
+  // 'ppc', 'ppc64', 's390', 's390x',
+  // 'x32', and 'x64'.
+  arch: string
+
+  download_url: string
+}
+
+export interface IToolRelease {
+  version: string
+  stable: boolean
+  release_url: string
+  files: IToolReleaseFile[]
+}
+
+export async function _findMatch(
+  versionSpec: string,
+  stable: boolean,
+  candidates: IToolRelease[]
+): Promise<IToolRelease | undefined> {
+  const archFilter = os.arch()
+  const platFilter = os.platform()
+
+  let result: IToolRelease | undefined
+  let match: IToolRelease | undefined
+
+  let file: IToolReleaseFile | undefined
+  for (const candidate of candidates) {
+    const version = candidate.version
+
+    debug(`check ${version} satisfies ${versionSpec}`)
+    if (
+      semver.satisfies(version, versionSpec) &&
+      (!stable || candidate.stable === stable)
+    ) {
+      file = candidate.files.find(item => {
+        debug(
+          `${item.arch}===${archFilter} && ${item.platform}===${platFilter}`
+        )
+
+        let chk = item.arch === archFilter && item.platform === platFilter
+        if (chk && item.platform_version) {
+          const osVersion = module.exports._getOsVersion()
```

Comment:
Since `_getOsVersion()` is heavy (launches a process (mac) or reads a file from disk (linux)), consider caching the result.

For example:

```ts
          if (osVersion === undefined) {
            osVersion = _getOsVersion()
          }
```